controllers/workspace/devworkspace_controller.go

```diff
@@ -307,6 +307,12 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 	serviceAcctName := serviceAcctStatus.ServiceAccountName
 	reconcileStatus.Conditions[devworkspace.WorkspaceServiceAccountReady] = ""
 
+	pullSecretStatus := provision.PullSecrets(clusterAPI)
+	if !pullSecretStatus.Continue{
+		return reconcile.Result{Requeue: pullSecretStatus.Requeue}, pullSecretStatus.Err
```

Comment:
Note that it will be exposed in Workspace status as WAITING FOR DEPLOYMENT, not sure if we should update devfile/api or introduce our own conditions instead

Reply:
I've added https://github.com/devfile/devworkspace-operator/pull/284/commits/a8cfd710ce0070eb3689da84bec4df210c91f3f6, but note that it will be exposed on workspace status only in case pull secrets failed from first try.